libraries/abstractions/common_io/include/iot_i2c.h

```diff
@@ -42,70 +42,69 @@
+    eI2CCompleted = IOT_I2C_SUCCESS,         /*!< I2C operation completed successfully. */
+    eI2CDriverFailed,                        /*!< I2C driver returns error during last operation. */
+    eI2CNackFromSlave = IOT_I2C_NACK,        /*!< Unexpected NACK is caught. */
+    eI2CMasterTimeout = IOT_I2C_BUS_TIMEOUT, /*!< I2C operation not completed within specified timeout. */
 } IotI2COperationStatus_t;
 
 /**
  * @brief I2C bus configuration
  */
 typedef struct IotI2CConfig
 {
-    uint32_t ulMasterTimeout;   /**<! Master timeout value in msec, to relinquish the bus if slave does not respond. This is an extended feature of SMBus which is not part of I2C protocol. */
-    uint32_t ulBusFreq;         /**<! Bus frequency/baud rate */
+    uint32_t ulMasterTimeout; /**<! Master timeout value in msec, to relinquish the bus if slave does not respond. This is an extended feature of SMBus which is not part of I2C protocol. */
+    uint32_t ulBusFreq;       /**<! Bus frequency/baud rate */
 } IotI2CConfig_t;
 
 /**
  * @brief Ioctl request types.
  */
 typedef enum
 {
-    eI2CSendNoStopFlag,     /**<! Set flag to not send stop after transaction */
-                            /**<! Default is always stop for every transaction */
-                            /**<! Flag will auto reset to stop after one transaction if you set no stop */
-    eI2CSetSlaveAddr,       /**<! All the operations use this slave address after it is set. */
-    eI2CSetMasterConfig,    /**<! Sets the I2C bus frequency and timeout using the struct IotI2CConfig_t,
-                             *   default speed is Standard */
-    eI2CGetMasterConfig,    /**<! Gets the I2C bus frequency and timeout set for the I2C master. */
-    eI2CGetBusState,        /**<! Get the current I2C bus status. Returns eI2CBusIdle or eI2CBusy */
-    eI2CBusReset,           /**<! Master resets the bus. */
-    eI2CGetTxNoOfbytes,     /**<! Get the number of bytes sent in write operation. */
-    eI2CGetRxNoOfbytes,     /**<! Get the number of bytes received in read operation. */
+    eI2CSendNoStopFlag,  /**<! Set flag to not send stop after transaction */
+                         /**<! Default is always stop for every transaction */
+                         /**<! Flag will auto reset to stop after one transaction if you set no stop */
+    eI2CSetSlaveAddr,    /**<! All the operations use this slave address after it is set. */
+    eI2CSetMasterConfig, /**<! Sets the I2C bus frequency and timeout using the struct IotI2CConfig_t, default speed is Standard mode. */
+    eI2CGetMasterConfig, /**<! Gets the I2C bus frequency and timeout set for the I2C master. */
+    eI2CGetBusState,     /**<! Get the current I2C bus status. Returns eI2CBusIdle or eI2CBusy */
+    eI2CBusReset,        /**<! Master resets the bus. */
+    eI2CGetTxNoOfbytes,  /**<! Get the number of bytes sent in write operation. */
+    eI2CGetRxNoOfbytes,  /**<! Get the number of bytes received in read operation. */
```

Comment:
Q:  What is the expectation on these two Get operations?  If I do a write, and then a read, is the expectation that GetTxNoOfbytes will still return the # of bytes from the write, or is the expectation that it is reset to zero at the beginning of each transaction?

Reply:
I think GetTxNoOfbytes or eI2CGetRxNoOfbytes will return the value in "last transaction".
For example,
- write -> no stop -> read: GetTx returns # of bytes in write and GetRx returns # of bytes in read
- write -> stop: GetTx returns # of bytes in write
- read -> stop: GetRx returns # of bytes in read

Other cases we can claim is undefined?

Reply:
Agreed.  My point is that we should document, and in your 3rd case of read->stop:  GetTx should return 0.  Since there were no Tx bytes.

Reply:
I see. So it means if users wants to know the # of bytes, they must call ioctl immediately after read/write, before next read/write happen. Is that what you think?

If there were a chance to re-design the interface, I would like read_sync and write_sync to return # of bytes (negative value is error).  Thus there will be fewer states in the library, making the behavior clearer.